VueJS/App.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <div id="app">
+    <Header></Header>
+    <AddTodo v-on:addTodo="addTodo"></AddTodo>
+    <Todos v-bind:todos="todos" v-on:del-todo="deleteTodo" v-on:edit-todo="editTodo"/>
+  </div>
+</template>
+
+<script>
+import Todos from './components/Todos'
+import AddTodo from './components/AddTodo'
+import Header from './components/layout/Header'
+
+
+export default {
+  name: 'app',
+  components: {
+    Header,
+    Todos,
+    AddTodo
+  },
+  data() {
+    return {
+      todos: [
+        { id: '1', message: 'Einkaufen', completed: false},
+        { id: '2', message: 'Aufräumen', completed: false},
+        { id: '3', message: 'Putzen', completed: false}
+      ],
+      nextTodoId: 4
+    }
+  },
+  methods: {
+    addTodo(newTodo) {
+      newTodo.id = this.nextTodoId++
+      this.todos.push(newTodo)
+    },
+    editTodo(id, newMessage) {
+      this.todos.filter(todo => {
+        if(todo.id == id) {
```

Comment:
```suggestion
        if(todo.id === id) {
```
This is one of the quirks of JavaScript :disappointed: see here:
https://codeburst.io/javascript-double-equals-vs-triple-equals-61d4ce5a121a